### PR TITLE
Add RE automation to Crossbow Crack Shot feat

### PR DESCRIPTION
**Pull request by CrimsonPolar**

### CODE REVIEW

**Comment by CrimsonPolar** [comment]
The proposed rule elements introduces a new toggleable RollOption called Crossbow Crack Shot. It uses the DamageRoll key to upgrade the damage die of simple crossbows in the player's inventory, and adds a +2 circumstance bonus to the damage roll of the attack.

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 1]
```diff
@@ -19,7 +19,38 @@
...
             "value": []
         },
-        "rules": [],
+        "rules": [
+            {
+                "domain": "damage-roll",
+                "key": "RollOption",
+                "option": "crossbow-crack-shot",
+                "toggleable": true,
+                "label": "Crossbow Crack Shot",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:28]: Shouldn't need this label.

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 2]
```diff
@@ -19,7 +19,38 @@
...
+                "value": true
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:29]: Omit this, as it should probably default to false.

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 3]
```diff
@@ -19,7 +19,38 @@
...
+            },
+            {
+                "key": "DamageDice",
+                "predicate": [
+                    "item:category:simple",
+                    "item:tag:crossbow",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:35]: item:base:crossbow

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 4]
```diff
@@ -19,7 +19,38 @@
...
+                ],
+                "override": {
+                    "upgrade": true
+                },
+                "selector": "bow-weapon-group-damage"
+            },
+            {
+                "key": "FlatModifier",
+                "predicate": [
+                    "item:tag:crossbow",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:46]: item:base:crossbow

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 5]
```diff
@@ -19,7 +19,38 @@
...
+                "key": "DamageDice",
+                "predicate": [
+                    "item:category:simple",
+                    "item:tag:crossbow",
+                    "crossbow-crack-shot"
+                ],
+                "override": {
+                    "upgrade": true
+                },
+                "selector": "bow-weapon-group-damage"
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:41]: crossbow-damage

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 6]
```diff
@@ -19,7 +19,38 @@
...
+            },
+            {
+                "key": "FlatModifier",
+                "predicate": [
+                    "item:tag:crossbow",
+                    "crossbow-crack-shot"
+                ],
+                "selector": "bow-weapon-group-damage",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:49]: crossbow-damage

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 7]
```diff
@@ -19,7 +19,38 @@
...
+                "key": "RollOption",
+                "option": "crossbow-crack-shot",
+                "toggleable": true,
+                "label": "Crossbow Crack Shot",
+                "value": true
+            },
+            {
+                "key": "DamageDice",
+                "predicate": [
+                    "item:category:simple",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:34]: Consider sorting the predicates.

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 8]
```diff
@@ -19,7 +19,36 @@
...
+                "key": "DamageDice",
+                "predicate": [
+                    "crossbow-crack-shot",
+                    "item:base:crossbow",
+                    "item:category:simple"
+                ],
+                "override": {
+                    "upgrade": true
+                },
+                "selector": "crossbow-damage"
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:39]: Actually, I think I mislead you and this should be “strike-damage”.

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 9]
```diff
@@ -19,7 +19,36 @@
...
+            },
+            {
+                "key": "FlatModifier",
+                "predicate": [
+                    "crossbow-crack-shot",
+                    "item:base:crossbow"
+                ],
+                "selector": "crossbow-damage",
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:47]: Same here.

File: packs/data/feats.db/crossbow-crack-shot.json (hunk 3 above)
Comment (stwlam) [reply packs/data/feats.db/crossbow-crack-shot.json]: It should in fact be "item:tag:crossbow"

File: packs/data/feats.db/crossbow-crack-shot.json [hunk 10]
```diff
@@ -19,7 +19,36 @@
...
+                "key": "DamageDice",
+                "predicate": [
+                    "crossbow-crack-shot",
+                    "item:category:simple",
+                    "item:tag:crossbow"
+                ],
+                "override": {
+                    "upgrade": true
+                },
+                "selector": "crossbow-damage"
```
Comment (simonward) [inline packs/data/feats.db/crossbow-crack-shot.json:39]: "selector": "strike-damage"

**Pull request merged**